.github/workflows/ci-pull-requests.yml

```diff
@@ -3,6 +3,20 @@ name: Test Pull Requests
 on: [pull_request]
 
 jobs:
+  check-commit-messages:
```

Comment:
is there no way to do this just for the squash/merge commit? Could we check the PR title instead since that's the default for the commit message?

Reply:
hey :wave: do we usually squash to one commit prior to merge? in some cases I guess it still makes sense to leave more than one commit in a PR and so we'd want to check all of the commits?

primary motivation for this was to ensure changelog generation works though, do you have any info on how that is implemented currently? I've been using the strategy 'check all commits (merge commits or otherwise) are in conventional commit format' as a proxy for ensuring this outcome :sweat_smile: 

if the changelog generation only looks at certain types of commits that could change things for sure :+1:

Reply:
yes we squash all commits into one commit.. the changelog is generated based on master commits.. so we don't care about commits in the feature branches.. only master

Reply:
@mutdmour I did some experimentation in a test repo with the [release process you pointed out](https://www.notion.so/n8n/Release-Process-fce65faea3d5403a85210f7e7a60d0f8#e91cea66bb774580b5fa0b073b2a3fc4) (thanks for that) and it does appear that conventional-changelog-cli is looking at non-merge-commits.

For example, even if you squash to one commit, and merge the PR, with a history on master like:
```
(merge commit with title from PR title) feat: fancy new feature
(normal commit from PR) my-non-conventional-commit-msg
...prior history...
```

This will not generate a changelog including `feat: fancy new feature`, in fact the changelog is empty.

But if you name normal commits conventionally as well, with a history on master like:
```
(merge commit with title from PR title) feat: fancy new feature
(normal commit from PR) feat: fancy new feature
...prior history...
```

The changelog is generated correctly.

So we'll need to lint non-merge-commit messages, not the PR title. This should be good for the times we want to review the git history as well.

I do want to stress that **you don't need to care about this until you're ready to merge**. If you used non-conventional commit messages, there'll be a failing check on the PR, but as soon as you squash to commit(s) with a conventional message, the check will start passing :+1: 

Let me know your thoughts, if you're happy I'll carry on implementing this to check all commit messages in a PR.

Reply:
thanks for testing it out directly. Let's try this out then.

Reply:
> If you used non-conventional commit messages, there'll be a failing check on the PR, but as soon as you squash to commit(s) with a conventional message, the check will start passing

I believe in this case the checks would never pass before one would actually merge, as the "Squash & merge" step is when GitHub actually squashes the commits together, making the check pass.

In other words, the only way to make the checks pass *before* merging would be to manually squash and force-push them. Which I isn't a bad thing as such, but might make the barrier of entry a bit higher for casual contributors...

That said, I'm all in for favor of adding this check! Didn't test drive it myself yet, but seems fine!

Reply:
@mieky you're quite right, @mutdmour just explained this to me on Slack as well, I mistakenly assumed people were locally using git rebase to clean up the feature branch, rather than the Github squash+merge feature. I'll need to rethink this, and maybe go back to the PR title linting idea instead.

Reply:
I wouldn't want to force every engineer in the company to start using git rebase, this may create it's own problems anyway.

Reply:
@agentreno Yeah, you're probably right! Linting the PR title alone would likely already tackle most of the issue, as the original problem was about a PR with an invalid commit message format not being included in the changelog.

Reply:
just to note, don't like git rebase.. because it loses the review history on Github.

Reply:
After some more testing with the squash+merge workflow and the changelog tool, concluded that @mutdmour was right first time round :sweat_smile: switched to PR title linting.